jackrabbit-api/src/main/java/org/apache/jackrabbit/api/binary/BinaryDownloadOptions.java

```diff
@@ -0,0 +1,317 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.jackrabbit.api.binary;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.osgi.annotation.versioning.ProviderType;
+
+/**
+ * Specifies the options to be used when obtaining a direct download URI via
+ * {@link BinaryDownload#getURI(BinaryDownloadOptions)}.  Setting these options
+ * allows the caller to instruct the service provider that these options should
+ * be applied to the response to a request made with the URI returned.
+ * <p>
+ * To specify download options, obtain a {@link BinaryDownloadOptionsBuilder}
+ * via the {@link #builder()} method, then specify the options desired and
+ * get the object via {@link BinaryDownloadOptionsBuilder#build()}.
+ * <p>
+ * If no options are needed, use {@link BinaryDownloadOptions#DEFAULT} which
+ * instructs the implementation to use the service provider default behavior.
+ */
+@ProviderType
+public final class BinaryDownloadOptions {
+    private final String mimeType;
+    private final String encoding;
+    private final String fileName;
+    private final String dispositionType;
+
+    private BinaryDownloadOptions(final String mimeType,
+                                  final String encoding,
+                                  final String fileName,
+                                  final String dispositionType) {
+        this.mimeType = mimeType;
+        this.encoding = encoding;
+        this.fileName = fileName;
+        this.dispositionType = dispositionType;
+    }
+
+    /**
+     * Provides a default instance of this class.  Using the default instance
+     * indicates that the caller is willing to accept the service provider
+     * default behavior.
+     */
+    public static final BinaryDownloadOptions DEFAULT =
+            BinaryDownloadOptions.builder().build();
+
+    /**
+     * Returns the MIME type that should be assumed for the binary that is to be
+     * downloaded.  This value should be a valid {@code jcr:mimeType}.  This
```

Comment:
"MIME" -> "Internet Media Type"

FWIW, maybe we should also not copy JCR's splitting of type and charset, and just provide the complete value for the Content-Type header field.